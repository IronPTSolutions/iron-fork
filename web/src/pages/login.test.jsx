import {
  cleanup,
  fireEvent,
  render,
  screen,
  waitFor,
} from "@testing-library/react";
import { beforeEach, describe, expect, test, vi } from "vitest";
import Login from "./login";
import { BrowserRouter } from "react-router-dom";
import { AlertProvider } from "../contexts/alert.context";
import AuthContext from "../contexts/auth.context";

const doLogin = vi.fn();

const LoginInWrapper = () => (
  <BrowserRouter>
    <AlertProvider>
      <AuthContext.Provider value={{ doLogin }}>
        <Login />
      </AuthContext.Provider>
    </AlertProvider>
  </BrowserRouter>
);

describe("<Login/>", () => {
  beforeEach(() => {
    cleanup();
  });

  test("render successfully", () => {
    // Given

    // When
    render(<LoginInWrapper />);

    // Then
    screen.getByText("Login");
    screen.getByLabelText("Email address");
    screen.getByLabelText("Password");

    expect(document.body).toMatchSnapshot();
  });

  test("happy case", async () => {
    // Given
    render(<LoginInWrapper />);

    // When

    const email = screen.getByLabelText("Email address");
    const password = screen.getByLabelText("Password");

    fireEvent.change(email, { target: { value: "john@example.com" } });
    fireEvent.change(password, { target: { value: "12345678" } });

    fireEvent.submit(screen.getByTestId("login-form"));

    // Then
    await waitFor(() => {
      expect(doLogin).toHaveBeenCalledWith({
        email: "john@example.com",
        password: "12345678",
      });
    });
  });
});
