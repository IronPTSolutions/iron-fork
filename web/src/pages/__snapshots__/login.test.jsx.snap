// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Login/> > render successfully 1`] = `
<body>
  <div>
    <div>
      <div
        class="container my-4"
      >
        <form
          data-testid="login-form"
        >
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="email"
            >
              Email address
            </label>
            <input
              class="form-control "
              id="email"
              name="email"
              required=""
              type="email"
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="password"
            >
              Password
            </label>
            <input
              class="form-control "
              id="password"
              name="password"
              required=""
              type="password"
            />
          </div>
          <button
            class="btn btn-success"
            type="submit"
          >
            Login
          </button>
        </form>
      </div>
    </div>
  </div>
</body>
`;
